yogstation/code/game/gamemodes/gangs/milestones.dm

```diff
@@ -0,0 +1,434 @@
+/datum/objective/gang  // Abstract
+
+/datum/objective/gang/dominate
+	name = "Dominate"
+	explanation_text = "Take over the station by activating the Dominator device."
+
+/datum/objective/gang/dominate/check_completion()
+	var/datum/team/gang/gang = team
+	return (gang.winner)
+
+/datum/objective/gang/control
+	name = "Control"
+	var/control_amount
+
+/datum/objective/gang/control/New()
+	control_amount = rand(15, 30)
+	explanation_text = "Control at least [control_amount] percent of the station at the end of the shift."
+
+/datum/objective/gang/control/check_completion()
+	var/datum/team/gang/gang = team
+	if(!(gang.territories.len >= control_amount))
+		return FALSE
+	return TRUE
+
+/datum/objective/gang/money
+	name = "Earn credits"
+	var/money_amount
+	var/datum/bank_account/gang_account
+
+/datum/objective/gang/money/New()
+	money_amount = rand(10000, 40000)
+	target_amount = money_amount
+	explanation_text = "Have at least [target_amount] credits in your gang's account at the end of the shift."
+
+/datum/objective/gang/money/check_completion()
+	var/datum/team/gang/gang = team
+	return (gang.registered_account.has_money(money_amount))
+
+/datum/objective/gang/members
+	name = "Convert gangsterts"
+
+/datum/objective/gang/members/New()
+	target_amount = rand(7,12)
+	explanation_text = "Have at least [target_amount] gangsters in your gang at the end of the shift."
+
+/datum/objective/gang/members/check_completion()
+	var/datum/team/gang/gang = team
+	gang.count_members()
+	return (gang.members.len >= target_amount)
+
+/datum/objective/gang/vip // Abstract type
+	var/target_role_type = FALSE
+	var/list/possible_targets = list()
+	var/list/datum/mind/officers
+	var/list/datum/mind/heads
+	var/list/blacklist = list()
+
+/datum/objective/gang/vip/inside_man/New()
+	name = "Inside Man"
+	find_inside_man()
+
+/datum/objective/gang/vip/inside_man/proc/find_inside_man()
+	for(var/datum/mind/possible_target in SSjob.get_living_sec())
+		possible_targets += possible_target
+	if(!possible_targets.len) // DEBUG_TEXT
```

Comment:
better error handling here please, you need to do *something* other than just saying "this failed"